src/dune_pkg/opam_solver.ml

```diff
@@ -769,6 +769,21 @@ let solve_lock_dir solver_env version_preference repos ~local_packages ~constrai
             stats.expanded_variables
             solver_env
         in
+        Package_name.Map.iter
+          pkgs_by_name
+          ~f:(fun { Lock_dir.Pkg.deps; info = { name; _ }; _ } ->
+            List.iter deps ~f:(fun (loc, dep_name) ->
+              if Package_name.Map.mem local_packages dep_name
+              then
+                User_error.raise
+                  ~loc
+                  [ Pp.textf
+                      "Dune does not support non-local dependencies depending on local \
```

Comment:
Instead of using local, how about "packages in the workspace cannot depend on packages outside the workspace". This way we don't need to introduce new terminology.

Reply:
updated